helix/package-principles.rst

```diff
@@ -0,0 +1,137 @@
+Appendix: Principles of Package and Component Design
+====================================================
+
+Helix is based on Robert Martin's "Principles of Package and Component Design" 
+(Chapter 28,  *Agile Principles, Patterns, and Practices in C#*, Prentice Hall, 2006). 
+These six principles set forth a conceptual framework for managing component boundaries ("cohesion") and interrelationships ("coupling").
+The following discussion presents these principles, provides a brief synopsis of how Martin presents them, and shows how they guide
+specific Helix practices.  Note that the original material uses the term "component," whereas Helix uses the term "module." We will use
+"component" when referring to packages in the original pre-Sitecore context, and we will use "module" when referring to Helix.
+
+The Principles of Component Cohesion
+------------------------------------
+Martin's first three principles deal with what goes into a component, and what stays out.
+
+.. note::
+
+    **The Reuse/Release Equivalency Principle (REP)**
```

Comment:
Can we do these as headers instead of Notes?

Reply:
No problem. If any of the generated output looks wonky, I'll follow up here. Otherwise, I'll chnage all of these over.

Reply:
Formatting looks fine on Github, so I'll mark this resolved.

Reply:
Also confirmed on build (see screen shots on following conversation.